components/match2/commons/match_group_input.lua

```diff
@@ -539,6 +564,7 @@ function MatchGroupInput.getCommonTournamentVars(obj, parent)
 		parent.liquipediatiertype,
```

Comment:
just an idea, but we probably could clean this function (`MatchGroupInput.getCommonTournamentVars`) up a bit into something like
```lua
---@param obj table
---@param parent table?
---@return table
function MatchGroupInput.getCommonTournamentVars(obj, parent)
	parent = parent or {}

	Array.forEach({
		'game',
		'icon',
		'icondark',
		'liquipediatier',
		'liquipediatiertype',
		'publishertier',
		'series',
		'shortname',
		'tickername',
		'patch',
	}, function(key)
		obj[key] = Logic.emptyOr(obj[key], parent[key], Variables.varDefault('tournament_' .. key))
	end)

	obj.tournament = Logic.emptyOr(obj.tournament, parent.tournament, Variables.varDefault('tournament_name'))
	obj.date = Logic.emptyOr(obj.date, parent.date)
	obj.mode = Logic.emptyOr(obj.mode, parent.mode)

	return obj
end
```

`tournament` doesn't follow the wiki var name pattern of the rest
mode doesn't try to use the wiki var at all
same for date

Reply:
Later on I want to make it not mutate the input, but leaving it for later when more/all wikis have been refactored